Document cartographer parameters for per RPLIDAR (#1097)
viamrobotics/docs: docs/services/slam/cartographer/_index.md

{{% alert title="REQUIREMENTS" color="tip" %}}

> Running `cartographer` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default cartographer settings, and the example config shown below (which uses the default cartographer settings), are for a RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for RPLIDAR A1.

Reviewer: [nit] Can you add links to the following?
* Default cartographer settings
* example config
* `config_params`

Otherwise it is difficult to follow what you're referring to.

[opt]
```suggestion
Running `cartographer` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default cartographer settings and the example config shown below (which uses the default cartographer settings) are nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.
```

Author: Sounds good! Will do.

Author: @kkufieta I suggested this change, let me know if it looks good to you? https://github.com/viamrobotics/docs/pull/1097#discussion_r1175753033

Reviewer: Looks good to me! :)

{{% alert title="REQUIREMENTS" color="tip" %}}

> Running `cartographer-module` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default cartographer settings and the example config shown below (which uses the default cartographer settings) are nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.

Author: ```suggestion
Running `cartographer-module` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default ['config_params'](#config_params) for the cartographer library, and the example robot config shown below (which uses the default config_params) show nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.
```

> Running `cartographer-module` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default ['config_params'](#config_params) for the cartographer library, and the example robot config shown below (which uses the default config_params) show nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.

Author: ```suggestion
Running `cartographer-module` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default ['config_params'](#config_params) for the cartographer library, and the example robot config shown below (which uses the default config_params), show nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.
```

> Running `cartographer-module` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default ['config_params'](#config_params) for the cartographer library, and the example robot config shown below (which uses the default config_params), show nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.

Author: ```suggestion
Running `cartographer-module` in Live Data Collection mode requires a [RPLIDAR A1](https://www.slamtec.com/en/Lidar/A1) or [RPLIDAR A3](https://www.slamtec.com/en/Lidar/A3) LIDAR scanning device. The default ['config_params'](#config_params) for the cartographer library, and the example robot config shown below (which uses the default 'config_params'), show nominal parameters one can use for an RPLIDAR A3. See the notes next to the 'config_params' for recommended settings for an RPLIDAR A1.
```

| `optimize_every_n_nodes` | How many trajectory nodes are inserted before the global optimization is run. | Optional | `3` | |
| `num_range_data` | Number of measurements in each submap. | Optional | `100` | |
| `missing_data_ray_length` | Replaces the length of ranges that are further than `max_range` with this value. | Optional | `25` | Nominally set to max length. |

> | `max_range` | Maximum range of valid measurements. | Optional | `25` | The default value is for RPLIDAR A3. For RPLIDAR A1, use `12`. |
> | `min_range` | Minimum range of valid measurements. | Optional | `0.2` | The default value is for RPLIDAR A3. For RPLIDAR A1, use `0.15`. |

Reviewer: [nit/opt] I would keep it more open without saying that the value is specified for the Rplidar A3 (since it could be true for other sensors, too). Something like this?

```suggestion
| `max_range` | Maximum range of valid measurements. | Optional | `25` | For an RPLIDAR A3, set this value to `25`. For an RPLIDAR A1, use `12`. |
| `min_range` | Minimum range of valid measurements. | Optional | `0.2` | For an RPLIDAR A3, set this value to `0.2`. For RPLIDAR A1, use `0.15`. |
```

Author: Sounds good! Thanks for the suggestion.
